Code/Mantid/Framework/WorkflowAlgorithms/src/ConvolutionFitSequential.cpp

```diff
@@ -0,0 +1,670 @@
+int ConvolutionFitSequential::version() const { return 1; }
+
+/// Algorithm's category for identification. @see Algorithm::category
+const std::string ConvolutionFitSequential::category() const {
+  return "Workflow\\MIDAS";
+}
+
+/// Algorithm's summary for use in the GUI and help. @see Algorithm::summary
+const std::string ConvolutionFitSequential::summary() const {
+  return "Performs a sequential fit for a convolution workspace";
+}
+
+//----------------------------------------------------------------------------------------------
+/** Initialize the algorithm's properties.
+ */
+void ConvolutionFitSequential::init() {
+  declareProperty(
+      new WorkspaceProperty<>("InputWorkspace", "", Direction::Input),
+      "The input workspace for the fit.");
+
+  auto scv = boost::make_shared<StringContainsValidator>();
+  auto requires = std::vector<std::string>();
+  requires.push_back("Convolution");
+  requires.push_back("Resolution");
+  scv->setRequiredStrings(requires);
+
+  declareProperty("Function", "", scv,
+                  "The function that describes the parameters of the fit.",
+                  Direction::Input);
+
+  std::vector<std::string> backType;
+  backType.push_back("Fixed Flat");
+  backType.push_back("Fit Flat");
+  backType.push_back("Fit Linear");
+
+  declareProperty("BackgroundType", "Fixed Flat",
+                  boost::make_shared<StringListValidator>(backType),
+                  "The Type of background used in the fitting",
+                  Direction::Input);
+
+  declareProperty(
+      "StartX", EMPTY_DBL(), boost::make_shared<MandatoryValidator<double>>(),
+      "The start of the range for the fit function.", Direction::Input);
+
+  declareProperty(
+      "EndX", EMPTY_DBL(), boost::make_shared<MandatoryValidator<double>>(),
+      "The end of the range for the fit function.", Direction::Input);
+
+  declareProperty("Temperature", EMPTY_DBL(),
+                  boost::make_shared<MandatoryValidator<double>>(),
+                  "The Temperature correction for the fit. If no there is no "
+                  "temperature correction, use 0.0",
+                  Direction::Input);
+
+  auto boundedV = boost::make_shared<BoundedValidator<int>>();
+  boundedV->setLower(0);
+
+  declareProperty("SpecMin", 0, boundedV, "The first spectrum to be used in "
+                                          "the fit. Spectra values can not be "
+                                          "negative",
+                  Direction::Input);
+
+  declareProperty("SpecMax", 0, boundedV, "The final spectrum to be used in "
+                                          "the fit. Spectra values can not be "
+                                          "negative",
+                  Direction::Input);
+
+  declareProperty("Convolve", true,
+                  "If true, the fit is treated as a convolution workspace.",
+                  Direction::Input);
+
+  declareProperty("Minimizer", "Levenberg-Marquardt",
```

Comment:
Ah sorry, I misread the next line and assumed it would be a `StringListValidator`.
What you have here is correct.